### PR TITLE
[DC-2003] Run participant validation as part of generate_metrics

### PR DESCRIPTION
**Pull request by nishanthpp93**
* [DC-2003] Move relevant scripts and modularize
* [DC-2003] Move .gitignore to root

[DC-2003]: https://precisionmedicineinitiative.atlassian.net/browse/DC-2003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DC-2003]: https://precisionmedicineinitiative.atlassian.net/browse/DC-2003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ksdkalluri — APPROVED** [review]
@nishanthpp93 This is mostly refactoring, I don't see any issues. I suggest we do a test run before merging this into develop branch.

**Review by jp3477 — APPROVED** [review]
@nishanthpp93 To my eye, this looks good to me as well! But I agree with @ksdkalluri that we should do some kind of test runs for the whole process to make sure it's all glued together.

**Comment by nishanthpp93** [comment]
@ksdkalluri @jp3477 I did some more refactoring, please take a look.

File: data_steward/constants/validation/participants/validate.py
Comment (jp3477) [inline data_steward/constants/validation/participants/validate.py:485]: I don't think the variables in this query match with the expectations of the `get_participant_validation_summary_query` function. `drc_dataset_id` here would need to be changed to `dataset_id` and `hpo_id` would need to be changed to `id_match_table`. Am I interpreting this correctly?

**Review by jp3477 — DISMISSED** [review]
Think I found something that can be changed. Will keep looking through.

Also, @nishanthpp93 did you find a good way to execute this in the test environment? I'd love to toy around with running it for validation.

File: data_steward/constants/validation/participants/validate.py
Comment (nishanthpp93) [reply data_steward/constants/validation/participants/validate.py:485]: You are right, nice catch. I wasn't using this anywhere since it is too generic, but this needs to be changed in case we do plan to use it.

**Comment by nishanthpp93** [comment]
@jp3477 Unfortunately I haven't. This is mostly refactoring so I'm just relying on the components already being tested.

File: data_steward/constants/validation/participants/validate.py
Comment (ksdkalluri) [inline data_steward/constants/validation/participants/validate.py:24]: @nishanthpp93 nit: I think these constants can be moved to the constants directory.

File: data_steward/validation/main.py
Comment (ksdkalluri) [inline data_steward/validation/main.py:396]: @nishanthpp93 so the participant validation runs only when there are no duplicates. Can we add messaging about the issue in results.html and is there a decision on how we are going to approach this?

Comment (nishanthpp93) [reply data_steward/validation/main.py:396]: Yeah, that makes sense. I didn't want to add the new section in results.html until we know what metrics should go there. I think EHR Ops will come up with metrics to figure out what needs to be shown to sites. Right now, there will be a logging message printed if it was skipped.

File: data_steward/constants/validation/participants/validate.py
Comment (nishanthpp93) [reply data_steward/constants/validation/participants/validate.py:24]: I just realized, this is in constants right now

File: data_steward/validation/main.py
Comment (nishanthpp93) [reply data_steward/validation/main.py:396]: I don't think the sites anticipate this yet. @jp3477 Should we start showing warnings in the results.html related to participant validation or hold off on it?

Comment (jp3477) [reply data_steward/validation/main.py:396]: @nishanthpp93 , we should hold off. We're still deciding how we would want to display it.

**Comment by nishanthpp93** [comment]
@ksdkalluri @jp3477 I've made the changes and rebased with develop. Let me know if anything else is needed to get this in.

**Pull request merged**
